rpc/rest/auth.go

```diff
@@ -20,7 +20,7 @@ import (
 	"strings"
 )
 
-func auth(h http.Handler, auth BasicHTTPAuthenticator) http.HandlerFunc {
+func authMiddleware(h http.Handler, auth BasicHTTPAuthenticator) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		user, pass, _ := r.BasicAuth()
 		if !auth.Authenticate(user, pass) {
```

Comment:
can we have a separate rate limiter for authentication - limiting failed authentication attempts, ideally per user (IP address?)

Reply:
Yeah, I think we can. I assume both should be configurable separately? 

*Per user* could work for rate limiting API access of logged in users and *per IP* address could work for rate limiting failed logins. What do you think? @rastislavszabo

Reply:
>  I assume both should be configurable separately?

Yeah, auth rate limiting usually will be much more strict

> Per user could work for rate limiting API access of logged in users and per IP address could work for rate limiting failed logins. 

great!